src/actions/Paste.ts

```diff
@@ -0,0 +1,107 @@
+import {
+  Action,
+  ActionPreferences,
+  ActionReturnValue,
+  Graph,
+  TypedSelection,
+} from "../Types";
+import displayPendingEditDecorations from "../editDisplayUtils";
+import { env, Selection } from "vscode";
+import { runForEachEditor } from "../targetUtils";
+import { computeChangedOffsets } from "../computeChangedOffsets";
+import { flatten, zip } from "lodash";
+import update from "immutability-helper";
+
+export default class Paste implements Action {
+  targetPreferences: ActionPreferences[] = [{ insideOutsideType: "inside" }];
+
+  constructor(private graph: Graph) {
+    this.run = this.run.bind(this);
+  }
+
+  async run([targets]: [TypedSelection[]]): Promise<ActionReturnValue> {
+    await displayPendingEditDecorations(
+      targets,
+      this.graph.editStyles.pendingModification0,
+      this.graph.editStyles.pendingLineModification0
+    );
+
+    const text = await env.clipboard.readText();
+
+    if (text.length === 0) {
+      throw new Error("Can't paste empty clipboard");
+    }
+
+    const lines = text.trim().split("\n");
+
+    const getText =
+      targets.length === lines.length
```

Comment:
I think we decided to match VSCode behaviour and check that number of lines is divisible by number of targets and if so split clipboard into blocks

Reply:
That's what line 37 does

Reply:
That checks equality, not divisibility, no? What am I missing?

Reply:
That is how vscode paste work.

Reply:
Nope!  Try it out.  Copy 4 lines, then make 2 cursors and paste.  It will split up the 4 lines into 2 blocks of 2

Reply:
Not on windows at least. I tried it multiple times before I made implementation and I tried it once again now.

Reply:
Haha wait really? That's really odd that that would differ on platforms. I'll try it one more time today; maybe I messed something up in my testing

Reply:
Yes. Really strange if it differs.